### PR TITLE
feat(HACBS-665): switch to using GitOps ApplicationSnapshot CRD

### PR DESCRIPTION
**Pull request by johnbieren**
This repository was temporarily maintaining an ApplicationSnapshot CRD
for testing purposes while the production one was developed by the
GitOps team. The production one is now ready to use, so this commit
removes the temporary one and switches over to using the production one.

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

File: controllers/release/release_adapter.go
Comment (johnbieren) [inline controllers/release/release_adapter.go:25]: @kasemAlem FYI this is how the new import looks

File: controllers/release/release_adapter.go
Comment (johnbieren) [inline controllers/release/release_adapter.go:173]: @kasemAlem FYI this is how the snapshot is referenced once this merges

File: tekton/pipeline_run.go
Comment (davidmogar) [inline tekton/pipeline_run.go:26]: Could we use the same name used in the main file? That's it, `gitopsv1alpha1`.

File: controllers/release/release_adapter.go
Comment (kasemAlem) [reply controllers/release/release_adapter.go:173]: @johnbieren Thanks, this was very helpful 
I've created a new branch to fix test ApplicationSnapshot part to be in GitOps api

File: tekton/pipeline_run.go
Comment (davidmogar) [reply tekton/pipeline_run.go:26]: I'm ok with you using `appstudioshared` as the GitOps service have API definitions in at least 3 different places, but please add the API version to the name (`appstudiosharedv1alpha1) so it's clear what version are we using.

This PR is blocking progress in e2e, so let's get it changed and merged as soon as possible.

Comment (johnbieren) [reply tekton/pipeline_run.go:26]: It is written exactly how the gitops team says to write it https://github.com/redhat-appstudio/managed-gitops/tree/main/appstudio-shared#remember-to-add-the-scheme-to-your-operators-maingo 
There is no reason to not follow their documentation

Comment (davidmogar) [reply tekton/pipeline_run.go:26]: My concerns are that the section in the README is only an example and not using a versioned import will only make more difficult to follow code specially during reviews. It will also require changes in the case of using multiple versions. In addition it breaks with the strategy followed up to this point in which every single CRD import is using a versioned name.

Having said that, let's move on to unblock progress on e2e and reevaluate in the future.

**Review by davidmogar — APPROVED** [review]

**Pull request merged**
